storage/remote/otlptranslator/prometheusremotewrite/otlp_to_openmetrics_metadata.go

```diff
@@ -42,36 +41,3 @@ func otelMetricTypeToPromMetricType(otelMetric pmetric.Metric) prompb.MetricMeta
 	}
 	return prompb.MetricMetadata_UNKNOWN
 }
-
-func OtelMetricsToMetadata(md pmetric.Metrics, addMetricSuffixes, allowUTF8 bool) []*prompb.MetricMetadata {
```

Comment:
Not in use.

Reply:
Converting metadata in `FromMetrics` is clearly more:

* Performant -> you don't need an extra loop over the OTLP data, you don't need to call `prometheustranslator.BuildCompliantName` twice, you don't need to make an extra slice to convert from `[]*prompb.MetricMetadata` to `[]prompb.MetricMetadata`.
* Readable/maintainable -> you need far less code by integrating into `FromMetrics`